packages/ui5-application-writer/src/types.ts

```diff
@@ -47,6 +47,7 @@ export interface UI5 {
     version: string;
     localVersion: string;
     typesVersion: string;
+    esmTypesVersion: string;
```

Comment:
When using typescript, we use the `esm` types, otherwise we are using the old experimental types, however, there is never a case where we need both, therefore, please use the property `typesVersion` and don't introduce a new one here.